src/WebJobs.Extensions.DurableTask/ContextImplementations/RemoteOrchestratorContext.cs

```diff
@@ -134,7 +134,7 @@ private void ThrowIfPlatformLevelException(FailureDetails failureDetails)
             // It's unclear if all OOMs are caught by the isolated process (probably not), and also if there are other platform-level
             // errors that are also caught in the isolated process and returned as a `OrchestratorExecutionResult`. Let's add them
             // to this method as we encounter them.
-            if (failureDetails.InnerFailure?.IsCausedBy<OutOfMemoryException>() ?? false)
+            if (failureDetails.IsCausedBy<OutOfMemoryException>())
```

Comment:
Just curious, which test failed before you made this change?

Reply:
The smoke test for Out of Memory exception retry handling - when David and I were writing that feature, every case we saw had the OOM wrapped at least one layer of InnerFailure deep, so we skipped checking the outermost failure details. Turns out, (likely when I upgraded the worker SDK version) this behavior changed and OOM became the outermost failure, meaning our check didn't catch it. Likely why we've had customers saying that the fix didn't help their app - because they were on latest worker SDK